druid-derive/src/attr.rs

```diff
@@ -105,52 +109,71 @@ impl Field {
         let mut same_fn = None;
         let mut lens_name_override = None;
 
-        for attr in field
-            .attrs
-            .iter()
-            .filter(|attr| attr.path.is_ident(BASE_ATTR_PATH))
-        {
-            match attr.parse_meta()? {
-                Meta::List(meta) => {
-                    for nested in meta.nested.iter() {
-                        match nested {
-                            NestedMeta::Meta(Meta::Path(path))
-                                if path.is_ident(IGNORE_ATTR_PATH) =>
-                            {
-                                if ignore {
-                                    return Err(Error::new(nested.span(), "Duplicate attribute"));
+        for attr in field.attrs.iter() {
+            if attr.path.is_ident(BASE_DRUID_DEPRECATED_ATTR_PATH) {
+                panic!("druid attribute is deprecated, please update it, it won't work")
```

Comment:
I'm not sure this will work if we haven't annotated the derives as accepting this attribute?

I'm also not sure `panic` is best, ideally we could return an error that displays over the span?

Reply:
Tested - at compile time this panic is shown as compilation error.
I'm thinking about splitting attribute parsing logic per derive(right now for derive Data we will parse for lens attribute as well, and then ignore it during code generation). We can write here more more explanations with examples if splitted.
We can make it warning, but I think for this case compile error is better.

Reply:
it does panic, but it doesn't show the correct span; it just shows the `Lens` derive. I was trying to describe something like this:
```suggestion
                return Err(Error::new(
                    attr.path.span(),
                    "The 'druid' attribute has been replaced with separate \
                    'lens' and 'data' attributes.",
                ));
```

Reply:
although I also agree this seems to let the main crate still try to compile, which creates more error messages than we need. I'll try and play around a bit more..

Reply:
okay actually I take this back, I think your solution is our best bet, until something like https://github.com/rust-lang/rust/issues/54140 lands.

Reply:
With this we have clear error messages with links to to place that should be fixed, I like it even more, didn't know that I can pass path to error to see it in a stacktrace.
I'm more concerned about decoupling attribute parsing for different derive attributes. But I can work on it in a separate branch.

Reply:
I have bad internet now days, didn't see new messages.